examples/src/main/scala/io/laserdisc/kafka/DangerousGDPR2.scala

```diff
@@ -0,0 +1,61 @@
+package io.laserdisc.kafka
+
+import org.apache.kafka.streams.scala.ImplicitConversions._
+import org.apache.kafka.streams.scala.Serdes._
+import org.apache.kafka.streams.scala.StreamsBuilder
+import org.apache.kafka.streams.scala.kstream._
+import org.apache.kafka.streams.{ StreamsConfig, Topology }
+import zio.UIO
+import zio.kafka.streams.KafkaStreamsTopology
+import zio.kafka.streams.testkit.ZTestTopology
+import zio.test.Assertion.equalTo
+import zio.test._
+import zio.test.environment.TestEnvironment
+
+import java.util.Properties
+
+object DangerousGDPR2 {
+  val config: Properties = {
+    val jProperties = new Properties()
+    jProperties.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092")
+    jProperties
+  }
+
+  val builder: StreamsBuilder           = new StreamsBuilder()
+  val newUsers: KStream[String, String] = builder.stream[String, String]("new-users-input")
+  val branches: Array[KStream[String, String]] =
+    newUsers.branch((k, _) => k.startsWith("u"), (k, _) => k.startsWith("c"))
+  val USERNAME                                            = 0
+  def securelyMappingUserForOutput(user: String): String  = identity(user)
+  val CREDIT_CARD                                         = 1
+  def securelyMappingCardForOutput(ccard: String): String = "************" + ccard.substring(12)
+  val processUser: KStream[String, String] =
+    branches(USERNAME).mapValues(username => securelyMappingUserForOutput(username))
+  val maskedCC: KStream[String, String] =
+    branches(CREDIT_CARD).mapValues(creditCard => securelyMappingUserForOutput(creditCard))
+  maskedCC.to("new-users-output")
```

Comment:
This error is a bit more hard to see because the functions have similar name. In this case we are writing credit card numbers without masking them.